mesonbuild/options.py

```diff
@@ -1056,11 +1056,7 @@ def replace(v: str) -> str:
 
         return changed
 
-    def set_option_from_string(self, keystr: T.Union[OptionKey, str], new_value: str) -> bool:
-        if isinstance(keystr, OptionKey):
-            o = keystr
-        else:
-            o = OptionKey.from_string(keystr)
+    def set_option_maybe_root(self, o: OptionKey, new_value: str) -> bool:
```

Comment:
nitpick: not sure `o` is a good variable name here... What about `key`?

Reply:
It is what was there before. :( I didn't want to add even more refactoring.